java/ql/src/experimental/Security/CWE/CWE-749/UnsafeAndroidAccess.ql

```diff
@@ -0,0 +1,139 @@
+/**
+ * @name Unsafe resource loading in Android webview
+ * @description JavaScript rendered inside WebViews can access any protected application file and web resource from any origin
+ * @kind path-problem
+ * @tags security
+ *       external/cwe/cwe-749
+ *       external/cwe/cwe-079
+ */
+
+import java
+import semmle.code.java.frameworks.android.Intent
+import semmle.code.java.frameworks.android.WebView
+import semmle.code.java.dataflow.FlowSources
+
+/**
+ * Allow universal access methods in the WebSettings class
+ */
+class AllowUniversalAccessMethod extends Method {
+  AllowUniversalAccessMethod() {
+    this.getDeclaringType() instanceof TypeWebSettings and
+    (
+      this.hasName("setAllowUniversalAccessFromFileURLs") or
+      this.hasName("setAllowFileAccessFromFileURLs")
+    )
+  }
+}
+
+/**
+ * `setJavaScriptEnabled` method for the webview
+ */
+class AllowJavaScriptMethod extends Method {
+  AllowJavaScriptMethod() {
+    this.getDeclaringType() instanceof TypeWebSettings and
+    this.hasName("setJavaScriptEnabled")
+  }
+}
+
+/**
+ * Check whether JavaScript is enabled in the webview with universal resource access
+ */
+predicate isJSEnabled(MethodAccess ma) {
```

Comment:
Or, probably cleaner to take a `Variable` here and do the `ma.getQualifier().getVariable()` at the callsite.

Reply:
Modified to take a <code>Variable</code>